aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/OrchestratorTest.java

```diff
@@ -156,7 +158,13 @@ public void preventConcurrentStateTransitions() throws AmplifyException {
         orchestrator.start();
 
         // Try to start it in a new thread.
-        new Thread(() -> orchestrator.start()).start();
+        new Thread(() -> {
+            try {
+                orchestrator.start();
+            } catch (DataStoreException exception) {
+                fail("Error occurred starting the orchestrator." + exception);
```

Comment:
This won't actually fail the test. jUnit will only report a test failure if the assertion error is raised on the test runner's thread. This will crash the `new Thread`, but it won't get reported.

If you _don't_ rely on the `fail` for the test to be useful, I'd suggest changing this `fail` to a comment, like:
```java
// This doesn't matter, we'll catch the failure later when we validate (thing you'll validate later).
```

If you _are_ expecting this `fail` to catch a bad state, you'd have to coalesce it back into control flow on the test runner thread, somehow.